File: parallel_execution/sources/collapse.2.f
! @@name:	collapse.2
! @@type:	F-fixed
! @@operation:	run
! @@expect:	success
! @@version:	omp_3.0
      program test
!$omp parallel
!$omp do private(j,k) collapse(2) lastprivate(jlast, klast)
      do k = 1,2
        do j = 1,3
          jlast=j
          klast=k
        enddo
      enddo
!$omp end do
!$omp single
      print *, klast, jlast !2, 3
!$omp end single
!$omp end parallel
      end program test

File: parallel_execution/sources/collapse.2.c
/*
* @@name:	collapse.2
* @@type:	C
* @@operation:	run
* @@expect:	success
* @@version:	omp_3.0
*/
#include <stdio.h>
int main()
{
   int j, k, jlast, klast;
   #pragma omp parallel
   {
      #pragma omp for collapse(2) lastprivate(jlast, klast)
      for (k=1; k<=2; k++)
         for (j=1; j<=3; j++)
         {
            jlast=j;
            klast=k;
         }
      #pragma omp single
      printf("%d %d\n", klast, jlast); //2 3
   }
}
